service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MessageContext.java

```diff
@@ -0,0 +1,24 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import org.springframework.core.NamedThreadLocal;
+
+public class MessageContext {
+
+    private static ThreadLocal<IdMapper> idMapperHolder = new NamedThreadLocal<>("IdMapper");
+
+    public static void resetMessageContext() {
+        idMapperHolder.remove();
+    }
+
+    public static void setMessageContext(IdMapper idMapper) {
```

Comment:
This is only ever called with new IdMapper().

Would it make sense to have an `initialiseMessageContext()` that hides creation of the IdMapper? That way we could add more threadlocals to the this class to capture different parts of the message context

Reply:
good idea. I noticed I could also create the mapper when fetching it for the first time.